src/server/hset_family.cc

```diff
@@ -808,6 +825,50 @@ void HSetFamily::HExists(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void HSetFamily::HExpire(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  string_view key = parser.Next();
+  string_view ttl_str = parser.Next();
+  uint32_t ttl_sec;
+  constexpr uint32_t kMaxTtl = (1UL << 26);
+  if (!absl::SimpleAtoi(ttl_str, &ttl_sec) || ttl_sec == 0 || ttl_sec > kMaxTtl) {
```

Comment:
Hm... They're similar, but not identical?
HExpire syntax is: `HEXPIRE key sec FIELDS 1 field1` while my understanding was that FieldExpire syntax was supposed to be `FIELDEXPIRE key sec field1`

So, it's possible to unify some of that, but I landed on just partly duplicating it, since it's actually not the same.

Reply:
> FIELDEXPIRE key sec field1 

I thought it's identical to `HEXPIRE`. IMO no matter how many times I search it on google, I can't seem to find it so if you have a link on that command I would be extremely happy.

As for the rest, I would just add a bool at the end of the common function `with_fields` and parse the keyword `FIELDS` on demand based on what the caller of the function decides (hexpire/fieldexpire)

Reply:
The [link to HExpire](https://redis.io/docs/latest/commands/hexpire/).
But you mean documentation for `FIELDEXPIRE`? I don't think that exists anywhere yet, and is just what @romange sketched out in https://github.com/dragonflydb/dragonfly/issues/3027

Reply:
no no I know what `hexpire` is, I could not find `fieldexpire` but now it makes sense. Then I would model it after `hexpire` and add the prerequisite of `FIELDS`. That way the parsing will be identical

Reply:
@NegatioN Hi! Could I sincerely ask you to `remove the FIELDS delimeter` from `FIELDEXPIRE` ? I synced internally and we decided to diverge them. Furthermore, on the implementation side, you can just add an extra `bool` that conditionally parses the `fields` member

Reply:
I'll try to join the parsing into one function with a bool to optionally include checking for `FIELDS N`.

Just to clarify: What do you mean by "Remove the FIELDS delimiter" from `FIELDEXPIRE` and `we decided to diverge them`?
To my eyes, `FIELDS N` were never a part of `FIELDEXPIRE`, so there's nothing to remove? But I might be misunderstanding you.

Reply:
My proposal was to make the two commands have a similar format but after discussing this internally we decided to keep them as is. That is, `FIELDS N` won't be part of `FIELDEXPIRE` and only part of `HEXPIRE`. So my end proposal is that since the parsing is only different for `FIELDS N` we can add a boolean to the parsing function and parse that field optionally. That way we have a common function that can be used by both of the commands

Reply:
> so there's nothing to remove? 

yes nothing to remove. Only merge the two parsing functions and parse the `FIELDS N` optinally by a bool argument